stratum/hal/lib/barefoot/bfrt_node.cc

```diff
@@ -138,6 +138,14 @@ ::util::Status BfrtNode::CommitForwardingPipelineConfig() {
   RETURN_IF_ERROR(bf_sde_interface_->AddDevice(device_id_, bfrt_config_));
 
   // Push pipeline config to the managers.
+  RETURN_IF_ERROR(p4runtime_bfrt_translator_->PushForwardingPipelineConfig(
+      bfrt_config_.programs(0).p4info()));
+
+  // Augment the P4Info so managers will use the original bitwith from P4 code
+  // for every fields (e.g, 9-bit port number instead of 32-bits).
+  ASSIGN_OR_RETURN(const auto& low_level_p4info,
+                   p4runtime_bfrt_translator_->GetLowLevelP4Info());
```

Comment:
Not sure how to name this. Basically, it is a P4Info with original bit width for match field, packet metadata, and action params.
The reason we need to pass this p4info is mostly that packet-io manager needs to use the original bit width to parse packet-io messages.